src/schedule.py

```diff
@@ -1,99 +1,123 @@
-            events[name] = []
-            for e in ini['events']:
-                events[name].append(ini['events'][e])
-        except KeyError as e:
-            logger.error('Corrupted ini file')
+    if len(date_array) > 2:
+        if date_array[0] == 'Mid':
+            q = date_array[1]
+        else:
+            q = date_array[2]
+    else:
+        q = date_array[0]
+
+    if q == 'Q1':
+        month = 3
+    elif q == 'Q2':
+        month = 6
+    elif q == 'Q3':
+        month = 9
+    elif q == 'Q4':
+        month = 12
+    else:
+        month = 1
 
-    create_ics(events, ics_dir)
+    date_object = None
+    if date_array[0] == 'Mid':
+        date_object = datetime(year, month - 1, 15, 0, 0)
+    elif date_array[0] == 'Start' and date_array[1] == 'of':
+        date_object = datetime(year, month - 2, 15, 0, 0)
+    else:
+        date_object = last_day_of_month(datetime(year, month, 1, 0, 0))
+    return date_object
+
+
+def generate_description(cards):
+    description = "Patch note:\n"
+    for card in cards:
+        description += "- %s: %s\n" % (card['name'], card['description'])
+    return description
+
+
+def parse_roadmap(roadmap):
+    events = []
+    for release in roadmap['releases']:
+        event = Event()
+        if release['released']:
+            begin_date = datetime.strptime(release['description'].split(' ', 1)[1], '%B %d, %Y')
+        else:
+            begin_date = get_date_from_quarter(release['description'])
+        event.begin = begin_date
+        event.make_all_day()
+        event.name = 'Star Citizen %s release' % release['name']
+        event.description = generate_description(release['cards'])
+        events.append(event)
+    return events
+
+
+def create_ics(events):
+    calendar = Calendar()
+    calendar.events = events
+    with open(config['core']['ics_output'], 'w') as ics_file:
+        ics_file.writelines(calendar)
+    return 0
+
+
+def mkschedule():
+    logger.info('Start mkschedule')
+    return_code = 0
+    try:
+        roadmap = get_roadmap()
+    except requests.HTTPError as e:
+        logger.error('Error while downloading the roadmap: %s', e)
+    except ValueError as e:
+        logger.error('Error while extracting the roadmap: %s', e)
+    except KeyError as e:
+        logger.error('Error while reading the roadmap: %s', e)
+
+    events = parse_roadmap(roadmap)
+    return_code = create_ics(events)
+    logger.info('End mkschedule')
+    return return_code
 
 
 if __name__ == '__main__':
-    argc = len(sys.argv)
-    if argc >= 3:
-        kw = {'level': logging.INFO, 'format': CORE['logging_format']}
-        if argc >= 4:
-            kw['filename'] = sys.argv[3]
-        logging.basicConfig(**kw)
-
-        logger.info('Start mkschedule')
-        mkschedule(ics_dir=sys.argv[1], img_dir=sys.argv[2])
-        logger.info('End mkschedule')
-    else:
-        print('Usage: {} ICS_DIR IMG_DIR [LOG_FILE]'.format(sys.argv[0]))
+    load_config('config.cfg')
```

Comment:
There's no `config.cfg` file in the repo anymore. Can you either add a sample one or the one you use?